helper/plan.php

```diff
@@ -623,7 +623,7 @@ protected function stepThroughNamespaces() {
         $lines = explode("\n", $lines);
 
         foreach($lines as $line) {
-            list($src, $dst) = explode("\t", trim($line));
+            list($src, $dst) = explode("\t", trim($line) . "\t");
```

Comment:
Why is this necessary? Is this because the root namespace is an empty string that is removed by trim? If this is the case, could we maybe not trim the tab character?

Reply:
Without this, the unit tests fail on PHP 8  with an undefined array key error (6 occurrences):
```
1) plugin_move_namespace_move_test::test_move_wiki_namespace
Undefined array key 1

/home/dregad/dev/dokuwiki/move/helper/plan.php:626
/home/dregad/dev/dokuwiki/move/helper/plan.php:395
/home/dregad/dev/dokuwiki/move/_test/namespace_move.test.php:61
```

The `list($src, $dst)` assignment expects an array with at least 2 items; if there is only one (i.e. if *$line* is a string without a tab character, then the error is triggered.

To be honest, I just fixed the symptoms without fully analyzing the root cause, so maybe this is not appropriate. 

Actually looking into this, the data read from the *namespaces* file into the $lines array has a trailing newline `wiki\tfoo:\n` so the explode() statement at line 623 creates an array with 2 items; the first line  is parsed correctly, but the second is an empty string that triggers the error.

So maybe the right fix would be to rtrim $lines before calling explode so this empty line is not processed.


 I just fixed the symptoms
, caused by